includes/shortcodes/class-sensei-shortcode-user-courses.php

```diff
@@ -64,7 +84,18 @@ class Sensei_Shortcode_User_Courses implements Sensei_Shortcode_Interface {
 	 * @param string $shortcode  the shortcode that was called for this instance
 	 */
 	public function __construct( $attributes, $content, $shortcode ) {
-
+		global $wp_query;
+		$this->page_id = $wp_query->get_queried_object_id();
+		if ( ! isset( $attributes['status'] ) && $this->is_my_courses() ) {
+			$this->in_my_courses_page = true;
+			// In My Courses page, let's overrride the setting.
+			if ( isset( $_GET[ self::MY_COURSES_STATUS_FILTER ] ) ) {
+				$course_filter_by_status = sanitize_text_field( $_GET[ self::MY_COURSES_STATUS_FILTER ] );
+				if ( ! empty( $course_filter_by_status ) && in_array( $course_filter_by_status, array( 'all', 'active', 'complete' ), true ) ) {
+					$attributes['status'] = $course_filter_by_status;
```

Comment:
I'm getting a warning here. It looks like WP can pass an empty string for `$attributes`. Looks like adding `shortcode_atts()` at the top of `__construct()` is the standard, which handles the cast and runs the `shortcode_atts_{$shortcode}` filter for that shortcode.